src/iso.c

```diff
@@ -0,0 +1,535 @@
+                iso_p->cur_offset -= ATAPI_BLOCKSIZE;
+                iso_p->cur_extent++;
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+/**
+ * @brief Lookup offset of the dir/file requested
+ * @details Finds the offset of the file or directory named by
+ * the full pathname starting at the root of the iso filesystem
+ * (no relative paths allowed).
+ *
+ * @param pname The string name of file or directory to lookup (requires leading /)
+ * @param dl Pointer to integer to fill in indicating dir/file data length
+ * @param ata_unit The ata_unit to search for the dir/file on
+ * @param is_dir_search 1 if we are looking up a directory, 0 if a file
+ * @return Offset (extent number) of the pathname, or -1 if not found or error
+ */
+static long int iso_look_up(const char *pname, uint32_t *dl, int ata_unit, bool is_dir_search) {
+    int root_dr_loc;
+    struct iso_point *iso_p = iso_media_open(ata_unit);
+    //locate the root directory
+    iso_media_seek(iso_p, ROOT_DR_OFFSET, SEEK_SET);
+
+    //Read root directory record from PVD
+    struct directory_record dr;
+    unsigned char loc_of_parent[8];
+    int success = get_directory_record(iso_p, &dr);
+    if (!success) {
+        iso_media_close(iso_p);
+        return -1;
+    }
+
+    memcpy(&loc_of_parent, dr.loc_of_ext, 8);
+    root_dr_loc = bendian_chars_to_int(loc_of_parent + 4, 4);
+
+    if (strcmp(pname, "/") == 0) {
+        iso_media_close(iso_p);
+        if (is_dir_search) {
+            return root_dr_loc;
+        }
+        else {  //Tried to open "/" as file
+            return -1;
+        }
+    }
+    else {
+        iso_media_seek(iso_p, root_dr_loc * ISO_BLOCKSIZE, SEEK_SET);
+        //moving forward, get rid of the first slash and give the iso_p we already seek'd for
+        long int result = iso_recursive_look_up(&pname[1], iso_p, dl, is_dir_search);
+        iso_media_close(iso_p);
+        return result;
+    }
+}
+
+/**
+ * @brief Lookup offset the dir/file requested
+ * @details Only call iso_lookup, which in turn calls iso_recursive_lookup.
+ * This recurses on the next directory in pathname
+ * by looking at the directory record given by the current
+ * disk offset, but does not handle the case of root.
+ *
+ * @param pname The string name of file or directory to lookup (with no leading /)
+ * @param iso_p Pointer to the iso point of the start of the directory extent to search.
+ * @param dl Pointer to integer to fill in indicating dir/file data length
+ * @param is_dir_search 1 if we are looking up a directory, 0 if a file
+ * @return The extent offset for the pname, or -1 if not found or on error
+ */
+static long int iso_recursive_look_up (const char *pname, struct iso_point *iso_p, uint32_t *dl, bool is_dir_search) {
+    int next_is_found = 0;
+
+    //Find the location of the next slash in pname
+    int next_slash_index = 0;
+    while(pname[next_slash_index] != '/' && next_slash_index < strlen(pname)) {
```

Comment:
Can we use strtok here to split apart the different pieces of the path, instead of writing our own?